src/node_info.rs

```diff
@@ -66,6 +66,17 @@ impl NodeContact {
         )
     }
 
+    pub fn from_non_contactable_and_socket_addr(
```

Comment:
the `NonContactable` type is mainly used as an expressive error, and if we are to receive an instance of this type in a public api we would need to make sure the Enr does not declare a socket address under the current ip mode, or otherwise we break the guarantee that the given `socket_addr` and `enr` are consistent (Either the same or None in the `enr`)

So in the meantime I suggest that this function is private. We can later improve this
```suggestion
    fn from_non_contactable_and_socket_addr(
```

Reply:
For now I removed the entire function. My feeling is that new talk request API will suggest whether a function like this is beneficial.